brut.apktool/apktool-lib/src/main/java/brut/androlib/ApkBuilder.java

```diff
@@ -44,470 +45,455 @@
 import java.util.*;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.logging.Logger;
-import java.util.zip.CRC32;
-import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 
 public class ApkBuilder {
     private final static Logger LOGGER = Logger.getLogger(ApkBuilder.class.getName());
 
-    private final AtomicReference<AndrolibException> mBuildError = new AtomicReference<>(null);
-    private final Config mConfig;
     private final ExtFile mApkDir;
-    private BackgroundWorker mWorker;
+    private final Config mConfig;
     private ApkInfo mApkInfo;
     private int mMinSdkVersion = 0;
-
-    private final static String APK_DIRNAME = "build/apk";
-    private final static String UNK_DIRNAME = "unknown";
-    private final static String[] APK_RESOURCES_FILENAMES = new String[] {
-        "resources.arsc", "AndroidManifest.xml", "res", "r", "R" };
-    private final static String[] APK_RESOURCES_WITHOUT_RES_FILENAMES = new String[] {
-        "resources.arsc", "AndroidManifest.xml" };
-    private final static String[] APP_RESOURCES_FILENAMES = new String[] {
-        "AndroidManifest.xml", "res" };
-    private final static String[] APK_MANIFEST_FILENAMES = new String[] {
-        "AndroidManifest.xml" };
+    private BackgroundWorker mWorker;
+    private final AtomicReference<AndrolibException> mBuildError = new AtomicReference<>(null);
 
     public ApkBuilder(ExtFile apkDir) {
-        this(Config.getDefaultConfig(), apkDir);
+        this(apkDir, Config.getDefaultConfig());
     }
 
-    public ApkBuilder(Config config, ExtFile apkDir) {
-        mConfig = config;
+    public ApkBuilder(ExtFile apkDir, Config config) {
         mApkDir = apkDir;
+        mConfig = config;
     }
 
-    public void build(File outFile) throws BrutException {
-        LOGGER.info("Using Apktool " + ApktoolProperties.getVersion() + " with " + mConfig.jobs + " thread(s).");
+    public void build(File outApk) throws AndrolibException {
+        if (mConfig.jobs > 1) {
+            mWorker = new BackgroundWorker(mConfig.jobs - 1);
```

Comment:
May you explain the subtraction of 1?

Reply:
Hey, I explained in the PR:
```
Also, fixes the expected behavior for -j/--jobs option, e.g. for "-j 1" it's expected to do everything on the main thread, so for j > 1 it's actually j-1 for smali/baksmali, and the last one is for the main thread (for aapt).
```
The main thread is also a thread so if we use batch processing on multiple APKs at once (e.g. with GNU parallel), then there is no way to turn off multithreading in apktool completely, cause it does 1 job for smali/baksmali, and the main thread is also busy doing the resources, i.e. still multithreading. This can cause the batch process to hit the set max load more often. This edit corrects it.